README.md

```diff
@@ -2,83 +2,119 @@
+  * [File fields](#file)
+  * [Geoip fields](#geoip)
+  * [Host fields](#host)
+  * [Kubernetes fields](#kubernetes)
+  * [Log fields](#log)
+  * [Network fields](#network)
+  * [Organization fields](#organization)
+  * [Process fields](#process)
+  * [Service fields](#service)
+  * [Source fields](#source)
+  * [URL fields](#url)
+  * [User fields](#user)
+  * [User agent fields](#user_agent)
 * [Use cases](#use-cases)
-* [Implementing ECS](#implementing-ecs)
-* [About ECS](#about-ecs)
-
-# <a name="fields"></a>Fields
-
-List of available ECS fields.
- * [Base fields](#base)
- * [Agent fields](#agent)
- * [Cloud fields](#cloud)
- * [Container fields](#container)
- * [Destination fields](#destination)
- * [Device fields](#device)
- * [Error fields](#error)
- * [Event fields](#event)
- * [File fields](#file)
- * [Geoip fields](#geoip)
- * [Host fields](#host)
- * [Kubernetes fields](#kubernetes)
- * [Log fields](#log)
- * [Network fields](#network)
- * [Organization fields](#organization)
- * [Process fields](#process)
- * [Service fields](#service)
- * [Source fields](#source)
- * [URL fields](#url)
- * [User fields](#user)
- * [User agent fields](#user_agent)
-
-## <a name="base"></a> Base fields
-
-The base set contains all fields which are on the top level without a namespace.
-
-These are fields which are common across all types of events.
+* [FAQ](#faq-ecs)
 
+## <a name="implementing-ecs"></a>Implementing ECS
 
-| Field  | Description  | Type  | Multi Field  | Example  |
-|---|---|---|---|---|
-| <a name="@timestamp"></a>`@timestamp`  | Timestamp when the event was created.<br/>For log events this is expected to be when the event was generated and not when it was read.<br/>Timestamp is a required field and must exist in all events.  | date  |   | `2016-05-23T08:05:34.853Z`  |
-| <a name="tags"></a>`tags`  | Tags is a list of keywords which are used to tag each event.  | keyword  |   | `["production", "env2"]`  |
-| <a name="labels"></a>`labels`  | Labels is an object which contains key/value pairs.<br/>Labels can be used to add additional meta information to events. Label should not contain nested objects and all values are stored as keyword.<br/>An example usage is the docker and k8s labels.  | object  |   | `{key1: value1, key2: value2}`  |
-| <a name="message"></a>`message`  | For log events the message field contains the log message.<br/>In other use cases the message field can be used to concatenate together different values which are then freely searchable. Or if multiple messages exist they can be combined here into one message.  | text  |   | `Hello World`  |
+**Guidelines**
+
+* The document MUST have the `@timestamp` field.
+* The [data type](https://www.elastic.co/guide/en/elasticsearch/reference/6.2/mapping-types.html) defined for an ECS field MUST be used.
+* It SHOULD have the field `event.version` to define which version of ECS it uses.
+* As many fields as possible should be mapped to ECS.
 
+**Writing fields**
```

Comment:
These are rules for ECS field names. To know these rules is only required when adding new fields. Should we move this to the contribution guide?

Reply:
I trust your judgement.  I like having the guidelines here because it summarizes the approach for naming fields.  In the benefits, we talk about improved ability to remember commonly used names and improved ability to remember field names that have been forgotten. IMO these guidelines contribute to the goals/benefits.